git/util.py

```diff
@@ -582,8 +582,16 @@ def _from_string(cls, string):
     @classmethod
     def _main_actor(cls, env_name, env_email, config_reader=None):
         actor = Actor('', '')
-        default_email = get_user_id()
-        default_name = default_email.split('@')[0]
+        user_id = None  # We use this to avoid multiple calls to getpass.getuser()
+
+        def default_email():
+            nonlocal user_id
+            if not user_id:
+                user_id = get_user_id()
+            return user_id
+
+        def default_name():
+            default_email().split('@')[0]
```

Comment:
Is `return` missing here?

Reply:
I was investigating an error in our CI and found this patch which is included `gitpython=3.10.0`

https://github.com/mlflow/mlflow/pull/3584/checks?check_run_id=1297036420#step:4:1677

Reply:
Guess I messed up on this one :)

Will provide a follow-up fix right away.

I am sorry for this :(

Reply:
no problem :) Thanks for the quick reponse!

Reply:
Thanks everyone. With the next PR to fix this one you will get a new patch release right away.

Reply:
Fixing it in https://github.com/gitpython-developers/GitPython/pull/1073.

Once again, sorry for the inconvenient